bundles/core/org.openhab.core.jsr223/src/main/java/org/openhab/core/jsr223/internal/engine/scriptmanager/ScriptUpdateWatcher.java

```diff
@@ -101,7 +101,12 @@ public void run() {
 					}
 				}
 
-				scriptManager.scriptsChanged(addedScripts, removedScripts, modifiedScripts);
+				try {
+					scriptManager.scriptsChanged(addedScripts, removedScripts, modifiedScripts);
+				}
+				catch (Exception ex) {
+					logger.error("Error during script change processing", ex);
```

Comment:
I agree that it doesn't impact platform stability given this fix for the thread exit, but it does indicate an actual (vs. potential) error in one of the JSR223 Rule instances. Also, the current log level is consistent with the log level for an exception thrown in the execute method of a Rule (RuleExecutionRunnable.java:38). That said, please confirm that you want the message at WARN level and I'll make the change. 

For future reference, is there any developer documentation to guide logger level usage?